Duplicate canonical Shiftr key found. Not allowing the same pattern appended key in spec

We are receiving Duplicate shiftr issue. Could you please help us to resolve.
Error :  java.lang.IllegalArgumentException: Duplicate canonical Shiftr key found : @(3,clone&(2,1))
How to write generalized spec for below requirement.
Input:
```
{
	"clone1_GCPerProIdenInfoPhyInfoStreet": "Street",
	"clone1_GCPerProIdenInfoPhyInfoAddLin2": "Address1",
	"clone1_GCPerProIdenInfoPhyInfoAddLin3": "Address3",
	"clone1_Physician": "Physician_10000",
	"clone2_Physician": "Physician_10000",
	"clone2_GCPerProIdenInfoPhyInfoStreet": "Address1",
	"clone2_GCPerProIdenInfoPhyInfoAddLin2": "addresdd"
}
```
Requirement: for above data "clone*_Physician" key contains value as "Physician_1000*" we have to clear the respective fields.
"Regex pattern"

spec:
```
[
  {
    "operation": "shift",
    "spec": {
      "clone*_Physician": {
        "Physician_1000*": {
          "clone&(2,1)_GCPerProIdenInfoPhyInfoStreet": "",
          "clone&(2,1)_GCPerProIdenInfoPhyInfoAddLin2": "",
          "clone&(2,1)_GCPerProIdenInfoPhyInfoAddLin3": ""
        }
      }
    }
  }
]
```
Spec screenshot attached due to regex pattern(* star) is not displaying in issue description block.
![joltspec](https://cloud.githubusercontent.com/assets/16236352/16077472/b457d786-3316-11e6-93ee-52cfdb5c3954.PNG)
.
Thanks in Advance..

Log attached.
com.bazaarvoice.jolt.exception.SpecException: JOLT Chainr encountered an exception constructing Transform className:com.bazaarvoice.jolt.Shiftr at index:12.
at com.bazaarvoice.jolt.chainr.instantiator.DefaultChainrInstantiator.hydrateTransform(DefaultChainrInstantiator.java:58)
at com.bazaarvoice.jolt.chainr.ChainrBuilder.build(ChainrBuilder.java:64)
at com.bazaarvoice.jolt.Chainr.fromSpec(Chainr.java:92)
at com.eapp.utility.jolt.JOLTTransformationUtil.convertJSONToJSON(JOLTTransformationUtil.java:39)
at com.eapp.service.integration.impl.IntegrationServiceImpl.invokeCaseExportService(IntegrationServiceImpl.java:6262)
at com.eapp.controller.FormsController.caseExportGC(FormsController.java:432)
at sun.reflect.GeneratedMethodAccessor44.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at org.springframework.web.method.support.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:219)
at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:132)
at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:104)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandleMethod(RequestMappingHandlerAdapter.java:745)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:686)
at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:80)
at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:925)
at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:856)
at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:936)
at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:838)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:647)
at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:812)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:728)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
at net.sf.ehcache.constructs.web.filter.GzipFilter.doFilter(GzipFilter.java:79)
at net.sf.ehcache.constructs.web.filter.Filter.doFilter(Filter.java:93)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:222)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123)
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:171)
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:99)
at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:936)
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:407)
at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1004)
at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:589)
at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:1852)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
at sun.reflect.GeneratedConstructorAccessor36.newInstance(Unknown Source)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
at com.bazaarvoice.jolt.chainr.instantiator.DefaultChainrInstantiator.hydrateTransform(DefaultChainrInstantiator.java:43)
... 41 more
Caused by: java.lang.IllegalArgumentException: Duplicate canonical Shiftr key found : @(3,clone&(2,1))
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.createChildren(ShiftrCompositeSpec.java:136)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.<init>(ShiftrCompositeSpec.java:75)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.createChildren(ShiftrCompositeSpec.java:127)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.<init>(ShiftrCompositeSpec.java:75)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.createChildren(ShiftrCompositeSpec.java:127)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.<init>(ShiftrCompositeSpec.java:75)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.createChildren(ShiftrCompositeSpec.java:127)
at com.bazaarvoice.jolt.shiftr.spec.ShiftrCompositeSpec.<init>(ShiftrCompositeSpec.java:75)
at com.bazaarvoice.jolt.Shiftr.<init>(Shiftr.java:483)
... 45 more

--- a/jolt-core/src/main/java/com/bazaarvoice/jolt/common/pathelement/AmpPathElement.java
+++ b/jolt-core/src/main/java/com/bazaarvoice/jolt/common/pathelement/AmpPathElement.java
@@ -69,6 +69,7 @@ public class AmpPathElement extends BasePathElement implements MatchablePathElem
         }
         if ( literal.length() > 0 ) {
             tok.add( literal.toString() );
+            canonicalBuilder.append( literal.toString() );
         }
 
         tok.trimToSize();

--- a/jolt-core/src/test/java/com/bazaarvoice/jolt/ShiftrTest.java
+++ b/jolt-core/src/test/java/com/bazaarvoice/jolt/ShiftrTest.java
@@ -56,6 +56,7 @@ public class ShiftrTest {
             {"objectToArray"},
             {"passNullThru"},
             {"passThru"},
+            {"pollaxman_218_duplicate_speclines_bug"},
             {"prefixDataToArray"},
             {"prefixedData"},
             {"prefixSoupToBuckets"},

--- a/jolt-core/src/test/resources/json/shiftr/pollaxman_218_duplicate_speclines_bug.json
+++ b/jolt-core/src/test/resources/json/shiftr/pollaxman_218_duplicate_speclines_bug.json
@@ -1,0 +1,34 @@
+{
+    "input" : {
+        "clone1_Physician" : "Physician_10000",
+        "clone2_Physician" : "Physician_10000",
+        "clone1_GCPerProIdenInfoPhyInfoStreet" : "Street",
+        "clone1_GCPerProIdenInfoPhyInfoAddLin2" : "Address1",
+        "clone1_GCPerProIdenInfoPhyInfoAddLin3" : "Address3",
+        "clone2_GCPerProIdenInfoPhyInfoStreet" : "Address1",
+        "clone2_GCPerProIdenInfoPhyInfoAddLin2" : "addresdd"
+    },
+    "spec" : {
+        // match clone1_Physician and clone2_Physician
+        "clone*_Physician" : {
+            // match the RHS of Physician_10000
+            "Physician_1000*" : {
+                // if we matched then grab the "number" of the physician and use that to process the address
+                "@(2,clone&(1,1)_GCPerProIdenInfoPhyInfoStreet)" : "Physician&(2,1).street",
+                "@(2,clone&(1,1)_GCPerProIdenInfoPhyInfoAddLin2)" : "Physician&(2,1).adderLin2",
+                "@(2,clone&(1,1)_GCPerProIdenInfoPhyInfoAddLin3)" : "Physician&(2,1).adderLin3"
+            }
+        }
+    },
+    "expected" : {
+        "Physician1" : {
+            "street" : "Street",
+            "adderLin2" : "Address1",
+            "adderLin3" : "Address3"
+        },
+        "Physician2" : {
+            "street" : "Address1",
+            "adderLin2" : "addresdd"
+        }
+    }
+}

